### PR TITLE
Script/CoS: full instance rewrite - partial merge

### PR DESCRIPTION
**Pull request by ccrs**
ref #17860

### CODE REVIEW

File: src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/boss_chrono_lord_epoch.cpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2008-2019 TrinityCore <https://www.trinitycore.org/>
+ * Copyright (C) 2008-2018 TrinityCore <https://www.trinitycore.org/>
  *
```
Comment (Aokromes) [inline src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/boss_chrono_lord_epoch.cpp]: date xd

Comment (Aokromes) [inline src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/boss_chrono_lord_epoch.cpp]: date xd

**Comment by ccrs** [comment]
I just cherry-picked, cleanup is incomming... ™️

**Comment by ccrs** [comment]
@Keader @Wyrserth I kindly request help in form of testing and ingame bug report 😃 
or else this wont be merged in another 4 years 💃

File: src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/boss_salramm_the_fleshcrafter.cpp [hunk 2]
```diff
@@ -124,7 +139,35 @@ class boss_salramm : public CreatureScript
...
+        AuraScript* GetAuraScript() const override
+        {
+            return new spell_salramm_steal_flesh_AuraScript();
+        }
+};
+
 void AddSC_boss_salramm()
 {
     new boss_salramm();
+    new spell_salramm_steal_flesh();
```
Comment (Sorikoff) [inline src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/boss_salramm_the_fleshcrafter.cpp]: Perhaps you could use new script register `RegisterSpellScript`.

**Comment by Wyrserth** [comment]
@jackpoz could you put this on the test server? Would be easier and more accurate to test if we can get a group going :P

**Comment by jackpoz** [comment]
coudl you check startup errors ? This is what I get (but I might have missed some):
>ScriptName 'npc_stratholme_fluff_undead' exists in database, but no core script found!

**Comment by jackpoz** [comment]
Please check also the warnings reported by CI tools.

I created a new realm on the test server called "Test", it shares auth and chararacter databases with the main realm but has its own world database.
Ping me if you need anything else (like updating it and such).

Please don't push force anymore so I can just git pull and rebuild.

File: sql/updates/world/3.3.5/9999_99_99_99_world_COT_STRATHOLME.sql [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+	(`creatureid` = 26528 AND `groupid` IN (0,5));
+
+-- Arthas/Mal'ganis RP data
+DELETE FROM `creature_text` WHERE `creatureid` IN (28167,28169);
+INSERT INTO `creature_text` (`creatureid`,`groupid`,`text`,`type`,`probability`,`BroadcastTextId`,`TextRange`,`emote`,`comment`) VALUES
+(28167, 0, "Prince Arthas, may the light be praised! Many people in the town have begun to fall seriously ill, can you help us?", 12, 100, 27547, 0, 1, "RP2_LINE_CITIZEN1"),
+(28167, 1, "What? This can't be!", 12, 100, 27549, 0, 274, "RP2_LINE_CITIZEN2"),
+(28169, 0, "Oh, no...", 12, 100, 27552, 0, 0, "RP2_LINE_RESIDENT1");
+UPDATE `creature_text` SET `emote`=1 WHERE
+	(`creatureid` = 26499 AND `groupid` IN (12,14)) OR
```
Comment (Wyrserth) [inline sql/updates/world/3.3.5/9999_99_99_99_world_COT_STRATHOLME.sql]: Tabs to spaces.

**Comment by Wyrserth** [comment]
Tested this in both normal and heroic mode. Three issues:

[ ] - Arcane Disruptor only works when the player uses it while standing on top of the suspicious crates, because the crate gameobject blocks LoS. Also DB warnings about non-existing creatures in spawn_group.

[This fixes both issues](https://gist.github.com/Wyrserth/12f96c0bc9f3118c13a62908d0ac7a47).

[ ] - The timed boss Infinite Corruptor did not appear for me, even though I reached Mal'Ganis in Heroic mode when the timer was saying 15 minutes left.

[ ] - When Chromie spawns for the third time (after killing Mal'Ganis) she cannot be interacted with (the mouse cursor changes, but right-clicking yields nothing). Also a DB error because the NPC is a quest giver but has no questgiver npcflag (should be set in creature_template and removed on spawn, added again when Chromie transforms from dragon to gnome).

Those are all the issues I could find. Everything else works great!

**Comment by Aokromes** [comment]
Arcane Disruptor only works when the player uses it while standing on top of the suspicious crates, because the crate gameobject blocks LoS.

this bug exist even without patch.

**Comment by Wyrserth** [comment]
Thanks Treeston for the original rewrite! (forgot to mention in the commit, sorry)

**Comment by ccrs** [comment]
Not a fan of removing the DEBUG Logs
also, next time ask before merging

**Comment by Wyrserth** [comment]
It looked good enough to get live feedback rather than keep it as a PR. I'll ask next time, sorry.

**Pull request merged**